integration/shareddata/shareddata.go

```diff
@@ -85,6 +86,31 @@ func AllProviders() []Provider {
 	return maps.Values(res)
 }
 
+// Providers are array of providers.
+type Providers []Provider
+
+// Remove specified providers and return remaining providers.
+func (ps Providers) Remove(removeProviderNames ...string) Providers {
+	res := make([]Provider, 0, len(ps))
+
+	for _, p := range ps {
+		keep := true
+
+		for _, name := range removeProviderNames {
+			if p.Name() == name {
+				keep = false
+				break
+			}
+		}
+
+		if keep {
```

Comment:
If we use `continue`, this check is not needed 
```suggestion
```

Reply:
I think this is still needed because we have nested loop :thinking: